concepts/box/about.md

```diff
@@ -1,3 +1,153 @@
+
+- When you have a type whose size can’t be known at compile time and you want to use a value of that type in a context that requires an exact size
+
+- When you have a large amount of data and you want to transfer ownership but ensure the data won’t be copied when you do so
+
+- When you want to own a value and you care only that it’s a type that implements a particular trait rather than being of a specific type
+
+A practical usage of `Box` is when you have multiple structs which you want to handle the same via a trait which they all implement.
+
+Let's say we have two structs. Each struct is of something that can be annoying. We can have each struct implement an Annoyance trait like so
+
+```rust
+struct WerewolfOfLondon {}
+struct Rooster {}
+
+trait Annoyance {
+    fn annoy(&self) -> &'static str;
+}
+
+impl Annoyance for WerewolfOfLondon {
+    fn annoy(&self) -> &'static str {
+        "ahhh-OOOOOOO!"
+    }
+}
+
+impl Annoyance for Rooster {
+    fn annoy(&self) -> &'static str {
+        "errr-errr-errr-errr-ERRRRRRRRR!"
+    }
+}
+```
+
+Now let's say we want a function that will give us an Annoyance based on the time of day that's passed into it. We can first try this, which we discover won't compile
+
+```rust
+enum TimeOfDay {
+    Midnight,
+    Morning,
+}
+
+//return something that implements Annoyance
+fn annoy_me(time_of_day: TimeOfDay) -> impl Annoyance {
+    match time_of_day {
+        TimeOfDay::Midnight => WerewolfOfLondon {},
+        TimeOfDay::Morning => Rooster {},
+    }
+}
+
+// `match` arms have incompatible types
+// expected struct `WerewolfOfLondon`, found struct `Rooster`
+// this is found to be of type `WerewolfOfLondon`
+// `match` arms have incompatible types
+// you could change the return type to be a boxed trait object
+// if you change the return type to expect trait objects, box the returned expressions
+
+```
+
+The compiler, in all of its helpfulness, tells us exactly what to do, which is to return a boxed trait object. If you don't know what a trait object is,
+a simple way to describe it is a way of returning a specific object via [dynamic dispatch][dynamic dispatch] from a function that could return any object
+implementing a specified trait. Since we need dynamic dispatch, we can try just putting a `dyn` in front of the trait name and see if we get away with it.
+
+```rust
+fn annoy_me(time_of_day: TimeOfDay) -> dyn Annoyance {
+    match time_of_day {
+        TimeOfDay::Midnight => WerewolfOfLondon {},
+        TimeOfDay::Morning => Rooster {},
+    }
+}
+
+// the size for values of type `dyn Annoyance` cannot be known at compilation time
+// the trait `Sized` is not implemented for `dyn Annoyance`
+// all local variables must have a statically known size
+// unsized locals are gated as an unstable feature
+```
+
+One of the errors is that the size for `dyn Annoyance` cannot be known at compile time. This is because annoy_me does not know what specific trait object
```

Comment:
```suggestion
One of the errors is that the size for `dyn Annoyance` cannot be known at compile time. This is because `annoy_me` does not know what specific trait object
```